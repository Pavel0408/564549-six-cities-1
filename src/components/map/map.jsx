import leaflet from "leaflet";
import PropTypes from "prop-types";
import React, {PureComponent} from "react";

export class Map extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    return <div id="map" style={{height: 823}}></div>;
  }

  componentDidMount() {
    const {offers} = this.props;
    const city = [52.38333, 4.9];
    const icon = leaflet.icon({
      iconUrl: `img/pin-map.svg`,
      iconSize: [30, 30]
    });

    const zoom = 12;
    try {
      const map = leaflet.map(`map`, {
        center: city,
        zoom,
        zoomControl: false,
        marker: true
      });
      map.setView(city, zoom);

      leaflet
        .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
          attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
        })
        .addTo(map);

      offers.map((offer) => {
        const offerCords = offer.coordinate;
        return leaflet
          .marker(offerCords, {icon})
          .addTo(map);
      });
    } catch (e) {
      return;
    }
  }
}

Map.propTypes = {
  offers: PropTypes.array.isRequired
};
