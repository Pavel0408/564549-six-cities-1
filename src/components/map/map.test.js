import React from 'react';
import renderer from 'react-test-renderer';

import {Map} from "./map";

const offersMock = [
  {name: `Beautiful & luxurious apartment at great location`,
    image: `img/apartment-01.jpg`,
    price: 120,
    rating: 5,
    isPremium: false,
    isFavorite: true,
    coordinate: [52.3909553943508, 4.85309666406198]
  },
  {
    name: `Wood and stone place`,
    image: `img/room.jpg`,
    price: 80,
    rating: 10,
    isPremium: true,
    isFavorite: false,
    coordinate: [52.369553943508, 4.85309666406198]
  }
];

describe(`testing Map render`, () => {
  it(`Map is render correctly`, () => {
    const tree = renderer.create(<Map
      offers={offersMock}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
