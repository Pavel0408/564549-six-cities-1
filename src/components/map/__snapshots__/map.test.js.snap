// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Map render Map is render correctly 1`] = `
<div
  id="map"
  style={
    Object {
      "height": 823,
    }
  }
/>
`;
